numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -927,7 +927,7 @@ NpyIter_NestedIters(PyObject *NPY_UNUSED(self),
         }
         for (i = 0; i < nested_naxes[inest]; ++i) {
             PyObject *v = PySequence_GetItem(item, i);
-            npy_intp axis;
+            int axis;
```

Comment:
Actually, this is already a bug, since not only are we assigning `PyInt_AsLong` to `npy_intp` (instead of `PyArray_PyIntAsIntp`, but we don't check for `PyErr_Occured` if the int doesn't fit in a long